api/dbadapters/bigquery.ts

```diff
@@ -296,27 +297,20 @@ export class BigQueryDbAdapter implements IDbAdapter {
   }
 
   public async getMetadata(target: dataform.ITarget): Promise<IBigQueryTableMetadata> {
-    const metadataResult = await this.pool
-      .addSingleTask({
-        generator: async () => {
-          try {
-            const table = await this.getClient(target.database)
-              .dataset(target.schema)
-              .table(target.name)
-              .getMetadata();
-            return table;
-          } catch (e) {
-            if (e && e.errors && e.errors[0] && e.errors[0].reason === "notFound") {
-              // if the table can't be found, just return null
-              return null;
-            }
-            // otherwise throw the error as normal
-            throw e;
-          }
-        }
-      })
-      .promise();
-    return metadataResult && metadataResult[0];
+    try {
```

Comment:
since we now have this method allowed for public callers (which is fine), i do think it should still live inside the promise pool. however, i would suggest extracting a private method (e.g. `getMetadataOutsidePromisePool` or something), which both `setMetadata` and `getMetadata` call, and keep the promise pool logic for `getMetadata`.